components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -406,110 +412,85 @@ protected void processAuthenticationResponse(HttpServletRequest request,
-                        tenantAwareUsername = user.getUsername();
-                        username = UserCoreUtil.addTenantDomainToEntry(tenantAwareUsername, user.getTenantDomain());
-                        userId = user.getUserID();
-                        userStoreDomain = user.getUserStoreDomain();
-                    }
-                } catch (OrganizationManagementException e) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("IdentifierHandler failed while trying to resolving user's resident org", e);
-                    }
-                    throw new AuthenticationFailedException(
-                            ErrorMessages.ORGANIZATION_MGT_EXCEPTION_WHILE_TRYING_TO_RESOLVE_RESIDENT_ORG.getCode(),
-                            e.getMessage(), User.getUserFromUserName(username), e);
-                } catch (UserStoreException e) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("IdentifierHandler failed while trying to authenticate", e);
-                    }
-                    throw new AuthenticationFailedException(
-                            ErrorMessages.USER_STORE_EXCEPTION_WHILE_TRYING_TO_AUTHENTICATE.getCode(), e.getMessage(),
-                            User.getUserFromUserName(username), e);
-                }
-            }
-        }
-
         Map<String, Object> authProperties = context.getProperties();
         if (authProperties == null) {
             authProperties = new HashMap<>();
             context.setProperties(authProperties);
         }
 
-        if (getAuthenticatorConfig().getParameterMap() != null) {
-            String validateUsername = getAuthenticatorConfig().getParameterMap().get("ValidateUsername");
-            if (Boolean.parseBoolean(validateUsername)) {
-                AbstractUserStoreManager userStoreManager;
-                // Check for the username exists.
-                try {
-                    int tenantId = IdentifierAuthenticatorServiceComponent
-                            .getRealmService().getTenantManager().getTenantId(tenantDomain);
-                    UserRealm userRealm = IdentifierAuthenticatorServiceComponent.getRealmService()
-                            .getTenantUserRealm(tenantId);
-
-                    if (userRealm != null) {
-                        userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
-
-                        // If the user id is already resolved from the multi attribute login, we can assume the user
-                        // exists. If not, we will try to resolve the user id, which will indicate if the user exists
-                        // or not.
-                        if (userId == null) {
-                            userId = userStoreManager.getUserIDFromUserName(tenantAwareUsername);
+        if (validateUsernameAdaptiveParam != null) {
```

Comment:
Add a comment mentioning the requirement of checking the not null here. 
i.e if it is null need to give the priority to the server config.

Reply:
Fixed